diesel/src/deserialize.rs

```diff
@@ -255,6 +256,153 @@ where
 #[doc(inline)]
 pub use diesel_derives::QueryableByName;
 
+/// Trait indicating that a record can be selected and queried from the database.
+///
+/// Types which implement `TableQueryable` represent the result of a SQL query. This
+/// does not necessarily mean they represent a single database table.
+///
+/// Diesel represents the return type of a query as the struct. The purpose of this
+/// trait is to provide name of select statement and use with QueryableByName.
+///
+/// This trait can be [derived](derive.QueryableByName.html)
+///
+/// # Examples
+///
+/// If we just want to map a query to our struct, we can use `derive`.
```

Comment:
These examples does not match what this trait is supposed to do. (Seems like they are just copied from `Queryable`?)

Same applies for nearly all documentation below.

Reply:
Yes, I haven't start documents stuff. Would review all documents later

Reply:
Basically done.